### ISSUE TITLE
Fire checklist 🔥

### ISSUE THREAD

**Issue by oed** [ceramicnetwork/js-ceramic#344]
# Checklist for Fire release
This is meant to be an exhaustive list of things we need to finish before we can launch the Fire livenet. It's mainly focused around the core protocol and things that relate to it. Please leave a comment if you find anything that's missing, or have questions about a particular item.

## js-ceramic
- [x] **DocIDs**
  - [x] DocID class implementation
  - [x] Remove `doctype` property from genesis record
- [x] **3ID finalization**
  - [x] Resolver backwards compatibility
  - [x] PublicKey ids should be deterministic (see https://github.com/w3c/did-core/issues/419)
  - [x] Use released did-jwt lib
- [x] **Log data structure solidification**
  - [x] `data` & `header` used in records, `content` and `metadata` used in docstate
  - ~~Implement logarithmic `refs` property for faster log loading~~ (postponed)
- [x] **Advanced queries**
  - [x] Create advanced queries CIP
  - [x] Implement HTTP api
  - ~~Implement pubsub queries~~ (postponed)
- [x]  **Misc.**
  - [x] Set `isUnique` to default to true
  - [x] Correctly handle version = 0 for documents
  - [x] Publish ANCHOR_META in pubsub topic (see [specs](https://github.com/ceramicnetwork/specs#anchor-metadata-messages))
  - [x] Rename `owners` to `controllers`
  - [x] Rename packages (remove `ceramic` prefix)
  - [x] Rename `head`  to `tip` everywhere
  - [x] Rename `account-link` to `caip10-link`
  - [x] Make sure `schema` is always stored with version
  - [x] Do garbage collection of documents which are not pinned
  - [x] When starting the node there should be a check for pending anchors for each pinned document
- [x] Testing
  - [x] Make sure that the network is well connected (dht and discovery of other nodes works well)
  - [x] When signing key has been rotated, make sure that old records are still valid (based on `kid`)
  - [x] When anchor is requested, Ceramic starts to poll CAS. If we restart Ceramic, will the poll continue?

## Ceramic Anchor Service
- [x] Validate that record being anchored is actually a valid update
- [x] Bloom filter
  - [x] Finalize spec
  - [x] Implement
- [x] Tree size limit
- [x] Optimize Merkle tree, use sorted array of CIDs instead of map
- [x] Test on mainnet and parameterize (anchor frequency)
- [x] Add limited retries for blockchain transactions
- [x] Create alarm for less than MIN funds threshold on Ceramic Anchor Service

## IDX
- [x] Align schemas with CIPs
  - [x] Rename "Crypto Account Links" -> "Crypto Accounts"
- [x] Create final schema deployments

**Comment by dazuck** [comment]
@oed I don't have much to add on this level. 

I think this is out of scope but just in case, the 2 things on my mind are: 
1. What kind of testing and monitoring is must-have by Fire (if we're not explicit, we don't prioritize or plan for it appropriately) 
2. Who is running nodes on Fire, just us or others too? If anyone else (even just our investors etc) what do we need to plan for around that. Not sure how this relates to @zachferland 's infra questions from Mindmeld..

**Comment by simonovic86** [comment]
`Infrastructure`:
- create alarm for less than MIN funds threshold on Ceramic Anchor Service

`Ceramic Anchor Service`:
- add limited retries for blockchain transactions
- configure rate limiting mechanism for CAS (malicious attacks)

`Testing`:
- when anchor is requested, Ceramic starts to poll CAS. If we restart Ceramic, will the poll continue?

**Comment by oed** [comment]
Thanks @simonovic86 updated!

**Comment by valmack** [comment]
Wow looking back at this checklist now we are so close!

**Comment by oed** [comment]
Everything on this checklist is done. Closing :tada:
